src/code_13870_len_6980.c

```diff
@@ -14,15 +14,37 @@ void mtx_ident_mirror_y(Matrix4f* mtx) {
 
 INCLUDE_ASM(s32, "code_13870_len_6980", clear_npcs);
 
-INCLUDE_ASM(s32, "code_13870_len_6980", init_npc_list);
+void init_npc_list(void) {
+    if (gGameStatusPtr[0]->isBattle == 0) {
+        gCurrentNpcListPtr = &gWorldNpcList;
+    } else {
+        gCurrentNpcListPtr = &gBattleNpcList;
+    }
+    D_8009A604 = 0;
+    D_800A0B94 = 1;
+}
 
-INCLUDE_ASM(s32, "code_13870_len_6980", func_8003857C);
+void func_8003857C(void) {
+    s32 temp_v0;
```

Comment:
If you swap the variable order, you can move the initializations into the same line as the declarations I think.